### DISCUSSION TITLE
OPA External Data Overload Option

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#373] [OPA and Rego] [answered]
Hi there,

I am currently reading https://www.openpolicyagent.org/docs/latest/external-data/#option-2-overload-input/ and I would like to understand more about Option 2 overload through examples. Do you happen to have an end-to-end example that includes caching and pulling data from a local source?

Thanks!

**Comment by anderseknert** [comment, marked as answer]
The client is responsible for compiling an input to OPA containing whatever data OPA might need to make a decision. I don't think the client caches the input in most cases, but if you're fetching data from something like a database, it might be a good idea.

**Reply by anderseknert** [reply]
See the contrib repo for some examples https://github.com/open-policy-agent/contrib
